admin/tabs/activity/tracking/detail/ActivityDetailModel.js

```diff
@@ -0,0 +1,139 @@
+import {FormModel} from '@xh/hoist/cmp/form';
+import {dateTimeCol, GridModel} from '@xh/hoist/cmp/grid';
+import {managed, HoistModel, XH} from '@xh/hoist/core';
+import {numberRenderer} from '@xh/hoist/format';
+import {Icon} from '@xh/hoist/icon/Icon';
+import {action, observable} from '@xh/hoist/mobx';
+import {wait} from '@xh/hoist/promise';
+
+@HoistModel
+export class ActivityDetailModel {
+
+    /** @member {ActivityTrackingModel} */
+    parentModel;
+    /** @member {GridModel} */
+    @managed gridModel;
+    /** @member {FormModel} */
+    @managed formModel;
+
+    @observable formattedData;
+
+    constructor({parentModel}) {
+        this.parentModel = parentModel;
+
+        this.gridModel = new GridModel({
+            sortBy: 'dateCreated|desc',
+            enableColChooser: true,
+            enableExport: true,
+            exportOptions: {
+                columns: 'ALL',
+                filename: `${XH.appCode}-activity-detail`
+            },
+            emptyText: 'Select a group on the left to see detailed tracking logs.',
+            columns: [
+                {
+                    field: 'impersonatingFlag',
+                    headerName: Icon.impersonate(),
+                    headerTooltip: 'Indicates if the user was impersonating another user during tracked activity.',
+                    excludeFromExport: true,
+                    resizable: false,
+                    align: 'center',
+                    width: 50,
+                    renderer: (v, {record}) => {
+                        const {impersonating} = record.data;
+                        return impersonating ?
+                            Icon.impersonate({
+                                asHtml: true,
+                                className: 'xh-text-color-accent',
+                                title: `Impersonating ${impersonating}`
+                            }) : '';
+                    }
+                },
+                {field: 'id', headerName: 'Entry ID', width: 100, align: 'right', hidden: true},
+                {field: 'username', ...usernameCol},
+                {field: 'impersonating', width: 140, hidden: true},
+                {field: 'category', width: 120},
+                {field: 'msg', headerName: 'Message', flex: true, minWidth: 120, autosizeMaxWidth: 400},
+                {field: 'data', flex: true, minWidth: 120, autosizeMaxWidth: 400, hidden: true},
+                {field: 'device', width: 100},
+                {field: 'browser', width: 100},
+                {field: 'userAgent', width: 100, hidden: true},
+                {
+                    field: 'elapsed',
+                    headerName: 'Elapsed',
+                    width: 120,
+                    align: 'right',
+                    renderer: numberRenderer({
+                        label: 'ms',
+                        nullDisplay: '-',
+                        formatConfig: {thousandSeparated: false, mantissa: 0}
```

Comment:
this code overly clunky for something that should be simple and was required 3 times in this PR.  I will create a ticket